themes/default/theme/stencil/src/components/choosable/choosable.tsx

```diff
@@ -96,6 +96,118 @@ export class Choosable {
         const values = this.values ? this.values.split(",").map(v => v.trim()) : [];
         const isActive = this.selection && (this.selection === this.value || values.includes(this.selection));
 
+        // Pages that get structured data (both Variant A and B)
+        const structuredDataPages = [
+            // Variant A pages
+            'authorization/roleassignment',
+            'containerservice/managedcluster', 
+            'resources/resourcegroup',
+            'dbforpostgresql/server',
+            // Variant B pages (also get structured data)
+            'managedidentity/userassignedidentity',
+            'network/virtualnetwork',
+            'network/subnet', 
+            'app/managedenvironment'
+        ];
+        
+        const needsStructuredData = typeof window !== 'undefined' && 
+            structuredDataPages.some(p => 
+                window.location.pathname.includes(`azure-native/api-docs/${p}`)
+            );
+        
+        // Apply structured data wrapper for language code blocks on specific pages
+        if (this.type === 'language' && needsStructuredData) {
+            const languageMap = {
+                'typescript': 'TypeScript',
+                'javascript': 'JavaScript',
+                'python': 'Python',
+                'go': 'Go',
+                'csharp': 'C#',
+                'java': 'Java',
+                'yaml': 'YAML'
+            };
+            
+            const langValue = this.value || this.values;
+            const mappedLanguage = languageMap[langValue] || langValue;
+            
+            // Extract resource metadata from URL
+            const getResourceMetadata = () => {
+                const pathname = typeof window !== 'undefined' ? window.location.pathname : '';
+                const match = pathname.match(/azure-native\/api-docs\/([^\/]+)\/([^\/]+)/);
+                
+                if (match) {
+                    const [_, service, resource] = match;
+                    // Format service and resource names for display
+                    const formatName = (str: string) => {
+                        // Handle special cases
+                        const specialCases = {
```

Comment:
A special cases block scares me. Why do we need this, and is there any chance this data is already available somewhere on this site?